From 315f0b261ec1488373386ccc53de0e9b0d7fb980 Mon Sep 17 00:00:00 2001
From: Stefan Priebe <stefan@prie.be>
Date: Mon, 7 Nov 2011 19:58:27 +0100
Subject: [PATCH] PowerScheduler: only log share message every 60s
 PowerScheduler: let PowerScheduler run every 10s instead of
 an endless loop

---
 .../PowerScheduler/Handlers/ActiveSharesHandler.cs |   18 +++++++++++++++---
 .../Plugins/PowerScheduler/PowerScheduler.cs       |    5 +++++
 2 files changed, 20 insertions(+), 3 deletions(-)

diff --git a/TvEngine3/TVLibrary/Plugins/PowerScheduler/Handlers/ActiveSharesHandler.cs b/TvEngine3/TVLibrary/Plugins/PowerScheduler/Handlers/ActiveSharesHandler.cs
index 92c1629..4ae4f5e 100644
--- a/TvEngine3/TVLibrary/Plugins/PowerScheduler/Handlers/ActiveSharesHandler.cs
+++ b/TvEngine3/TVLibrary/Plugins/PowerScheduler/Handlers/ActiveSharesHandler.cs
@@ -194,6 +194,8 @@ namespace TvEngine.PowerScheduler.Handlers
     #region Variables
 
     private bool _enabled = false;
+    private bool _laststandbyreturnvalue = false;
+    DateTime _lastprinttime = DateTime.MinValue;
     private List<ShareMonitor> _sharesToMonitor = new List<ShareMonitor>();
 
     private ManagementObjectSearcher _searcher = new ManagementObjectSearcher(
@@ -302,13 +304,23 @@ namespace TvEngine.PowerScheduler.Handlers
             {
               if (shareBeingMonitored.Equals(connection))
               {
-                Log.Debug("{0}: Standby cancelled due to connection '{1}:{2}' on share '{3}'", HandlerName,
-                          connection.UserName, connection.ComputerName, connection.ShareName);
+                if (_laststandbyreturnvalue == false || DateTime.Now.Subtract(_lastprinttime).TotalSeconds > 60)
+                {
+                  Log.Debug("{0}: Standby cancelled due to connection '{1}:{2}' on share '{3}'", HandlerName,
+                            connection.UserName, connection.ComputerName, connection.ShareName);
+                  _lastprinttime = DateTime.Now;
+                  _laststandbyreturnvalue = true;
+                }
                 return true;
               }
             }
           }
-          Log.Debug("{0}: have not found any matching connections - will allow standby", HandlerName);
+          if (_laststandbyreturnvalue == true || DateTime.Now.Subtract(_lastprinttime).TotalSeconds > 60)
+          {
+            Log.Debug("{0}: have not found any matching connections - will allow standby", HandlerName);
+            _lastprinttime = DateTime.Now;
+            _laststandbyreturnvalue = false;
+          }
           return false;
         }
         return false;
diff --git a/TvEngine3/TVLibrary/Plugins/PowerScheduler/PowerScheduler.cs b/TvEngine3/TVLibrary/Plugins/PowerScheduler/PowerScheduler.cs
index 365337a..1f04945 100644
--- a/TvEngine3/TVLibrary/Plugins/PowerScheduler/PowerScheduler.cs
+++ b/TvEngine3/TVLibrary/Plugins/PowerScheduler/PowerScheduler.cs
@@ -843,7 +843,12 @@ namespace TvEngine.PowerScheduler
               SendPowerSchedulerEvent(PowerSchedulerEventType.Elapsed);
             }
             else
+            {
               CheckForStandby(false);
+              //Thread.Sleep(10000);
+              // STEFAN: it is enough to call it every 10s
+              _stopThread.WaitOne(10000);
+            }
           }
           catch (Exception ex)
           {
-- 
1.7.8.msysgit.0

